src/test/java/org/jenkinsci/plugins/workflow/cps/SnippetizerTester.java

```diff
@@ -103,6 +114,47 @@ protected Object invokeStep(StepDescriptor d, Object args) {
         r.assertEqualDataBoundBeans(step, actual);
     }
 
+    /**
+     * Recurses through the model of a {@link Describable} class's {@link DescribableModel} and its parameters to verify that doc generation will work.
+     *
+     * @param describableClass
+     *     A {@link Class} implementing {@link Describable}
+     * @throws Exception
+     *     If any errors are encountered.
+     */
+    @SuppressWarnings("unchecked")
+    public static void assertDocGeneration(Class<? extends Describable> describableClass) throws Exception {
+        DescribableModel<?> model = new DescribableModel(describableClass);
+
+        assertNotNull(model);
+
+        recurseOnModel(model);
+
+    }
+
+    private static void recurseOnTypes(ParameterType type) throws Exception {
+        if (type instanceof ErrorType) {
+            throw ((ErrorType)type).getError();
+        }
+
+        if (type instanceof ArrayType) {
+            recurseOnTypes(((ArrayType)type).getElementType());
+        } else if (type instanceof HomogeneousObjectType) {
+            recurseOnModel(((HomogeneousObjectType) type).getSchemaType());
+        } else if (type instanceof HeterogeneousObjectType) {
+            for (Map.Entry<String, DescribableModel<?>> entry : ((HeterogeneousObjectType) type).getTypes().entrySet()) {
+                recurseOnModel(entry.getValue());
+            }
+        }
```

Comment:
By design - any other type is a no-op in the docs generation from our perspective, in that they can't contain other types, and any error would have been encountered on entering the type itself.